packages/nodes-base/nodes/Webhook/Webhook.node.ts

```diff
@@ -99,10 +100,27 @@ export class Webhook extends Node {
 			ignoreBots: boolean;
 			rawBody: Buffer;
 			responseData?: string;
+			accessControl: IDataObject;
 		};
 		const req = context.getRequestObject();
 		const resp = context.getResponseObject();
 
+		const webhookName = context.getWebhookName();
+
+		if (webhookName === 'options') {
+			const { allowMethods, allowOrigin, maxAge } = options.accessControl || {};
+			const headers = {
+				'Access-Control-Allow-Methods': (allowMethods as string) || '*',
+				'Access-Control-Allow-Origin': (allowOrigin as string) || '*',
+				'Access-Control-Max-Age': String((maxAge as number) * 1000) || '60000',
+			};
+
+			resp.writeHead(200, headers);
+			return {
+				noWebhookResponse: true,
+			};
```

Comment:
This code needs to be in core, as CORS pre-flight requests are handled [there](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/WebhookHelpers.ts#L106), and they never reach this code.